If-CI-checks-fail-on-your-PR.md

```diff
@@ -1,42 +1,71 @@
+
+1. In GitHub Actions, select the job under 'Jobs' to see the logs for that particular job.
+
+1. Read the GitHub Actions log to figure out what the actual failure is.
+
+     ![Screenshot of GitHub Actions CI logs page](images/githubActionsLogs.png)
+
+1. Search for the error message on the [issue tracker](https://github.com/oppia/oppia/issues) to see whether the error has been reported before. If there's an open issue for the same error, see [What to do if the failure is due to an existing issue](#what-to-do-if-the-failure-is-due-to-an-existing-issue), below.
+
+1. Otherwise, the error might be due to your PR, and you'll need to investigate it further. See [Figuring out whether the failure is due to your PR or an existing issue](#figuring-out-whether-the-failure-is-due-to-your-pr-or-an-existing-issue), below.
+
+### Why is an unrelated CI check failing on my PR?
+
+The tests/checks on Oppia's develop branch sometimes fail incorrectly for two reasons:
+
+  * Sometimes, the tests in develop are "flaky", which means they pass sometimes and fail sometimes, even though the code has not changed. This is usually due to "race" conditions where multiple changes happen at the same time and it's not guaranteed which one happens first. The dev workflow team is trying to reduce cases like this, though they still happen occasionally.
+
+  * If an incorrect PR was recently merged to develop, this could cause the tests to fail. If you see the same failure happening on the [develop branch](https://github.com/oppia/oppia/actions?query=branch%3Adevelop), please follow the [[Revert and Regression Policy|Revert-and-Regression-Policy]] immediately so that we can revert the problematic changes as soon as possible. Otherwise, this will start to affect all the PRs and block the development workflow completely.
+
+### Failures in the merge queue
+
+If the CI checks pass on your PR, but you only see a failure in the merge queue when attempting to merge, do the following:
+
+1. Verify that the issue is not due to your PR (see [Figuring out whether the failure is due to your PR or an existing issue](#figuring-out-whether-the-failure-is-due-to-your-pr-or-an-existing-issue), below).
+
+1. Follow the instructions in [What to do if the failure is due to an existing issue](#what-to-do-if-the-failure-is-due-to-an-existing-issue), below (i.e. ensure that an issue is filed for the failure, and leave a comment on it).
+
+1. Leave a comment on your PR that (a) provides a link to the issue above, and (b) explicitly ask the maintainers to force-merge your PR.
 
 
 ## Figuring out whether the failure is due to your PR or an existing issue
 
+If there isn't an open issue already corresponding to the failure you are seeing, then either your code could be wrong, or the test/check on Oppia's develop branch could be incorrect. You will need to determine which of these cases it is.
```

Comment:
there could also be an error on develop outside the test/check code

Reply:
Thanks, fixed.